### PR TITLE
Fix raspberries 4 issue with rustup

### PR DESCRIPTION
**Pull request by nicofrand**
## Problem

- Rustup auto-detection of armfh is incorrect and the installation fails on raspberry 4.

## Solution

- Use yunohost's detection and specify the toolchain accordingly

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

File: scripts/install [hunk 1]
```diff
@@ -134,7 +134,15 @@ ynh_exec_as $app virtualenv --python=python3 --system-site-packages "${final_pat
 	source "${final_path}/venv/bin/activate"
 	set -o nounset
 
-	ynh_exec_warn_less ynh_exec_as $app RUSTUP_HOME="$final_path"/.rustup CARGO_HOME="$final_path"/.cargo bash -c 'curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -q -y'
+	# Rustup has an issue with the autodetection of armfh
+	# see https://github.com/rust-lang/rustup/issues/3342
+	rustup_toolchain="stable"
+	if [ $YNH_ARCH == "armfh" ];
+	then
```
Comment (Thovi98) [inline scripts/install:141]:
```suggestion
	if [ $YNH_ARCH == "armhf" ];
	then
```

File: scripts/upgrade [hunk 2]
```diff
@@ -165,7 +165,15 @@ ynh_exec_as $app virtualenv --python=python3 --system-site-packages "${final_pat
 	source "${final_path}/venv/bin/activate"
 	set -o nounset
 
-	ynh_exec_warn_less ynh_exec_as $app RUSTUP_HOME="$final_path"/.rustup CARGO_HOME="$final_path"/.cargo bash -c 'curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -q -y'
+	# Rustup has an issue with the autodetection of armfh
+	# see https://github.com/rust-lang/rustup/issues/3342
+	rustup_toolchain="stable"
+	if [ $YNH_ARCH == "armfh" ];
+	then
```
Comment (Thovi98) [inline scripts/upgrade:172]:
```suggestion
	if [ $YNH_ARCH == "armhf" ];
	then
```

**Review by Thovi98 — COMMENTED** [review]
Seems to be a typo?

I don’t have a raspberry to test but looks good otherwise

**Comment by nicofrand** [comment]
> Seems to be a typo?
> 
> I don’t have a raspberry to test but looks good otherwise

Lol I wonder how the fix could work then...
Good catch!

**Pull request closed without merging**